plugins/com.google.cloud.tools.eclipse.aeri/src/com/google/cloud/tools/eclipse/aeri/ExceptionSender.java

```diff
@@ -0,0 +1,120 @@
+
+  private static final String REPORT_URL = "https://clients2.google.com/cr/report";
+
+  private static final String NONE_MARKER = "__NONE__";
+
+  private static final String PRODUCT_KEY = "product";
+  private static final String VERSION_KEY = "version";
+  private static final String EXCEPTION_INFO_KEY = "exception_info";
+  private static final String COMMENTS_KEY = "comments";
+
+  private static final String ECLIPSE_BUILD_ID_LABEL = "eclipseBuildId: ";
+  private static final String JAVA_VERSION_LABEL = "javaVersion: ";
+  private static final String OS_LABEL = "os: ";
+  private static final String USER_SEVERITY_LABEL = "userSeverity: ";
+  private static final String USER_COMMENT_LABEL = "userComment: ";
+
+  private final String endpointUrl;
+
+  ExceptionSender() {
+    this(REPORT_URL);
+  }
+
+  @VisibleForTesting
+  ExceptionSender(String endpointUrl) {
+    this.endpointUrl = endpointUrl;
+  }
+
+  void sendException(IThrowable exception, String eclipseBuildId, String javaVersion,
+      String os, String osVersion, String userSeverity, String userComment) {
+    Map<String, String> parameters = new HashMap<>();
+    parameters.put(PRODUCT_KEY, CloudToolsInfo.EXCEPTION_REPORT_PRODUCT_ID);
+    parameters.put(VERSION_KEY, CloudToolsInfo.getToolsVersion());
+    parameters.put(EXCEPTION_INFO_KEY, formatStackTrace(exception));
+
+    String extraInfo = ECLIPSE_BUILD_ID_LABEL + nullOrEmptyToNone(eclipseBuildId) + "\n"
+        + JAVA_VERSION_LABEL + nullOrEmptyToNone(javaVersion) + "\n"
+        + OS_LABEL + os + " " + nullOrEmptyToNone(osVersion) + "\n"
+        + USER_SEVERITY_LABEL + nullOrEmptyToNone(userSeverity) + "\n"
+        + USER_COMMENT_LABEL + nullOrEmptyToNone(userComment);
+    // "comments" seems to be the only viable place where we can put extra info.
+    parameters.put(COMMENTS_KEY, extraInfo);
+
+    try {
+      // Internal system expects the product ID and version in URL too.
+      String urlParameters = "?product=" + CloudToolsInfo.EXCEPTION_REPORT_PRODUCT_ID
+          + "&version=" + CloudToolsInfo.getToolsVersion();
+      int code = HttpUtil.sendPostMultipart(endpointUrl + urlParameters, parameters);
+      if (code != HttpURLConnection.HTTP_OK) {
+        logger.log(Level.WARNING, "Failed to send exception report. HTTP response code: " + code);
+      }
+    } catch (IOException ex) {
+      logger.log(Level.WARNING, "Failed to send exception report.", ex);
+    }
+  }
+
+  private static String nullOrEmptyToNone(String string) {
+    if (string == null || string.trim().isEmpty()) {
+      return NONE_MARKER;
+    }
+    return string;
+  }
+
+  /** Format the modeled stack trace. */
+  @VisibleForTesting
+  static String formatStackTrace(IThrowable exception) {
+    if (exception == null) {
+      return NONE_MARKER;
+    }
+
+    // We need this manual formatting because exception of type "Exception" isn't anonymized.
```

Comment:
No code to do anonymization here. I will update the comment to clarify. What I am saying is like this:

"We should send a stack trace from AERI's `IThrowable`, not a stack trace from the standard Java `Throwable`, because `Throwable` is never anonymized while `IThrowable` might have been. And because AERI's `IThrowable` doesn't implement an equivalent of `Throwable.printStackTrace()`, we have to manually implement it."